gapis/api/vulkan/query_timestamps.go

```diff
@@ -544,4 +545,50 @@ func (t *queryTimestamps) PostLoop(ctx context.Context, out transform.Writer) {
 	for _, queryPoolInfo := range t.queryPools {
 		t.GetQueryResults(ctx, cb, out, queryPoolInfo)
 	}
+	t.cleanup(ctx, out)
+}
+
+func (t *queryTimestamps) cleanup(ctx context.Context, out transform.Writer) {
+	s := out.State()
+	cb := CommandBuilder{Thread: 0, Arena: s.Arena}
+
+	// Destroy query pool created in this transform.
+	for _, info := range t.queryPools {
+		cmd := cb.VkDestroyQueryPool(
+			info.device,
+			info.queryPool,
+			memory.Nullptr,
+		)
+		out.MutateAndWrite(ctx, api.CmdNoID, cmd)
+	}
+	t.queryPools = make(map[VkQueue]*queryPoolInfo)
+
+	// Free commandbuffer and commandpool allocated in this transform.
+	for commandPoolkey, commandPool := range t.commandPools {
```

Comment:
You would have to check the spec, but IIRC you can just destroy the command-pool, you dont have to destroy each individual command buffer. But this is fine too.

Reply:
Updated to just released the commandpool